### PR TITLE
Share option

### PR DESCRIPTION
**Pull request by aberllin**
- when user presses Share, the modal Share your todoList appears and the background window get blurred
- added email validation for the input, if email is not valid the error text under the email appears
- by pressing Cancel or X the modal disappear

Netlify - https://deploy-preview-8--optimistic-fermat-7ff6ec.netlify.app/

### CODE REVIEW

File: src/components/AppWindow/MenuComponents/MenuButton.tsx [hunk 1]
```diff
@@ -1,36 +1,55 @@
...
 export const MenuButton = () => {
-    const [showMenu, setShowMenu ] = useState(false)
+  const [showMenu, setShowMenu] = useState<boolean>(false)
+  const [showShareModal, setShowShareModal] = useState<boolean>(false)
 
-    const openMenu = () => {
-        setShowMenu(prev => !prev)
-    }
+  const openMenu = () => {
+    setShowMenu((prev) => !prev)
```
Comment (ruslanakholod) [inline src/components/AppWindow/MenuComponents/MenuButton.tsx]: just     setShowMenu(true)

File: src/components/AppWindow/MenuComponents/MenuOptions/ShareModal.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value)
+  }
+
+  const onSubmit = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    event.preventDefault()
+
+    if (!inputValue) {
+      setErrors('Email required')
+    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(inputValue)) {
```
Comment (ruslanakholod) [inline src/components/AppWindow/MenuComponents/MenuOptions/ShareModal.tsx]: const isValid  = !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(inputValue)

File: src/App.tsx [hunk 3]
```diff
@@ -23,7 +23,7 @@ function App() {
   
   useEffect(() => {
     const inStorage = localStorage.getItem('theme')
-    const initialValue: any = inStorage ? inStorage : 'light'
+    const initialValue: any = inStorage ? inStorage : Theme.light
```
Comment (arevej) [inline src/App.tsx:26]: Type `any`?

File: src/components/AppWindow/MenuComponents/MenuButton.tsx [hunk 4]
```diff
@@ -1,36 +1,55 @@
...
-import {MenuModal} from './MenuModal'
+import React, { useState } from 'react'
+import { HiOutlineDotsCircleHorizontal } from 'react-icons/hi'
+import { MenuModal } from './MenuModal'
 import styled from 'styled-components'
-
-
-
+import { ShareModal } from './MenuOptions/ShareModal'
 export const MenuButton = () => {
```
Comment (arevej) [inline src/components/AppWindow/MenuComponents/MenuButton.tsx:6]: How is this a reusable component? According to the task it should be  Add a Menu button to each App with an ability to provide different menu options"

File: src/components/AppWindow/MenuComponents/MenuModal.tsx [hunk 5]
```diff
@@ -1,30 +1,66 @@
...
 import styled from 'styled-components'
 
+interface Props {
+  open: boolean
+  onClose: () => void
+  openShareModal: () => void
+}
+
+export const MenuModal = ({ open, onClose, openShareModal }: Props) => {
+  const modalRef = React.createRef<HTMLInputElement>()
```
Comment (arevej) [inline src/components/AppWindow/MenuComponents/MenuModal.tsx:11]: Why not useRef?

File: src/components/AppWindow/MenuComponents/MenuModal.tsx [hunk 6]
```diff
@@ -1,30 +1,66 @@
...
+    (e: Event) => {
+      if (modalRef.current && modalRef.current.contains(e.target as Node)) {
+        return
+      }
+      if (open) return onClose()
+    },
+    [modalRef, onClose, open],
+  )
+
+  useEffect(() => {
```
Comment (arevej) [inline src/components/AppWindow/MenuComponents/MenuModal.tsx:22]: It would be nice to have it as a separate hook something like useOutsideClick

**Review by arevej — COMMENTED** [review]
You need to think about separating TodoList app ShareModal from MenuButton since MenuButton is a reusable/common component and ShareModal is TodoList app only. 

Let's make a meeting to discuss it more.

**Pull request closed without merging**